projects/hello_firmware/src/main.c

```diff
@@ -0,0 +1,49 @@
+#include <stdint.h>  // for integer types
+
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for logging
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for wait function
+
+#define COUNTER_DELAY_MS 500
+
+typedef struct Counter {
+  uint8_t counters[2];
+} Counter;
+
+typedef struct TimerData {
+  Counter main_counter;
+  bool incrementBoth;
+} TimerData;
+
+static void prv_increment_and_print_counter(int counter_num, Counter *main_counter) {
+  main_counter->counters[counter_num]++;
+  LOG_DEBUG("Counter %c: %d\n", counter_num + 65, main_counter->counters[counter_num]);
```

Comment:
I would prefer you use `'A'` instead of `65` to convert 0/1 to A/B, since in C `'A' == 65` and the character is clearer. I think it's okay for this though.